gcp_variant_transforms/transforms/merge_headers.py

```diff
@@ -13,100 +13,23 @@
-    """
-    self._resolver = resolver
-
-  def merge(self, first, second):
-    # type: (vcf_header_io.VcfHeader, vcf_header_io.VcfHeader) -> None
-    """Updates ``first``'s headers with values from ``second``.
-
-    If a specific key does not already exist in a specific one of ``first``'s
-    headers, that key and the associated value will be added. If the key does
-    already exist in the specific header of ``first``, then the value of that
-    key will be updated with the value from ``second``.
-
-    Args:
-      first: The VcfHeader object.
-      second: The VcfHeader object that's headers will be merged into the
-        headers of first.
-    """
-    if (not isinstance(first, vcf_header_io.VcfHeader) or
-        not isinstance(first, vcf_header_io.VcfHeader)):
-      raise NotImplementedError
-    self._merge_header_fields(first.infos, second.infos)
-    self._merge_header_fields(first.filters, second.filters)
-    self._merge_header_fields(first.alts, second.alts)
-    self._merge_header_fields(first.formats, second.formats)
-    self._merge_header_fields(first.contigs, second.contigs)
-
-  def _merge_header_fields(
-      self,
-      first,  # type: Dict[str, OrderedDict[str, Union[str, int]]]
-      second  # type: Dict[str, OrderedDict[str, Union[str, int]]]
-      ):
-    # type: (...) -> None
-    """Modifies ``first`` to add any keys from ``second`` not in ``first``.
-
-    Args:
-      first: first header fields.
-      second: second header fields.
-    Raises:
-      ValueError: If the header fields are incompatible (e.g. same key with
-        different types or numbers).
-    """
-    for second_key, second_value in second.iteritems():
-      if second_key not in first:
-        first[second_key] = second_value
-        continue
-      first_value = first[second_key]
-      if first_value.keys() != second_value.keys():
-        raise ValueError('Incompatible header fields: {}, {}'.format(
-            first_value, second_value))
-      merged_value = OrderedDict()
-      for first_field_key, first_field_value in first_value.iteritems():
-        second_field_value = second_value[first_field_key]
-        try:
-          resolution_field_value = self._resolver.resolve_attribute_conflict(
-              first_field_key,
-              first_field_value,
-              second_field_value)
-          merged_value.update({first_field_key: resolution_field_value})
-        except ValueError as e:
-          raise ValueError('Incompatible number or types in header fields:'
-                           '{}, {} \n. Error: {}'.format(
-                               first_value, second_value, str(e)))
-
-      first[second_key] = merged_value
-
 class _MergeHeadersFn(beam.CombineFn):
   """Combiner function for merging VCF file headers."""
 
-  def __init__(self, header_merger):
-    # type: (_HeaderMerger) -> None
+  def __init__(self, _header_merger):
```

Comment:
s/_header_merger/header_merger

Reply:
This name collides with the module name. Renamed to merger instead (as I assume the issue is that I have `_` in front of the param name, defining it as private).